src/Features/Core/Portable/SolutionCrawler/IdleProcessor.cs

```diff
@@ -65,9 +66,23 @@ protected async Task WaitForIdleAsync()
                     return;
                 }
 
-                // TODO: will safestart/unwarp capture cancellation exception?
-                var timeLeft = BackOffTimeSpanInMS - diffInMS;
-                await Task.Delay(Math.Max(MinimumDelayInMS, timeLeft), this.CancellationToken).ConfigureAwait(continueOnCapturedContext: false);
+                var blockedOnCompletion = notifyBlockedOnCompletion.BlockedOnCompletion;
+                using (var combinedCancellationToken = CancellationToken.CombineWith(blockedOnCompletion))
+                {
+                    try
+                    {
+                        // TODO: will safestart/unwarp capture cancellation exception?
+                        var timeLeft = BackOffTimeSpanInMS - diffInMS;
+                        await Task.Delay(Math.Max(MinimumDelayInMS, timeLeft), combinedCancellationToken.Token).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException) when (blockedOnCompletion.IsCancellationRequested && !CancellationToken.IsCancellationRequested)
+                    {
+                        // The delay terminated early to accommodate a blocking operation. Make sure to delay long
+                        // enough that low priority (on idle) operations get a chance to be triggered.
+                        await Task.Delay(TimeSpan.FromMilliseconds(10)).ConfigureAwait(false);
```

Comment:
if we don't have this waht happen?

Reply:
I would also add that if blockedOnCompletion.IsCancellationToken.IsCancelled, then return right away rather than do any of waiting.

Reply:
> if we don't have this waht happen?

The containing loop spins until WaitForAllOperationsAsync times out. I'm not sure what was getting blocked.

Reply:
Can you expand on this comment with the additional comments above?

Reply:
Expanded to the degree known.